keanu-project/src/main/java/io/improbable/keanu/vertices/dbl/nonprobabilistic/operators/unary/SigmoidVertex.java

```diff
@@ -17,22 +14,16 @@
      * @param inputVertex the vertex
      */
     public SigmoidVertex(DoubleVertex inputVertex) {
-        super(inputVertex.getShape(), inputVertex);
-    }
-
-    @Override
-    protected DoubleTensor op(DoubleTensor a) {
-        return a.unaryMinus().expInPlace().plusInPlace(1).reciprocalInPlace();
+        super(inputVertex,
+            a -> a.unaryMinus().expInPlace().plusInPlace(1).reciprocalInPlace(),
+            a -> op(a));
     }
 
-    @Override
-    public DualNumber calculateDualNumber(Map<Vertex, DualNumber> dualNumbers) {
-        DualNumber dualNumber = dualNumbers.get(inputVertex);
-        DoubleTensor x = dualNumber.getValue();
+    public static DualNumber op(DualNumber a) {
```

Comment:
Can we move this to the `DualNumber` class as a `sigmoid` function?

Reply:
Why? Would it get called anywhere else?

Reply:
It's just a convention we have, keeps the OpVertex fairly light and means you can the `DualNumber` class for any operations!